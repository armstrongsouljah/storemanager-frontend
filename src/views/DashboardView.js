/* eslint-disable no-undef */
/* eslint-disable no-shadow */
/* eslint-disable indent */
/* eslint-disable react/prefer-stateless-function */
import React, { Component, } from "react";
import { Redirect, } from "react-router-dom";
import { connect, } from "react-redux";

import Dashboard from "../components/Dashboard";
import Navigation from "../components/Navigation";
import getProducts from "../actions/getProducts";
import CornerMenu from "../components/CornerMenu";
import UserAddView from "./UserAddView";



export class DashboardView extends Component {
    state = {
      redirect: false,
      open: false,
      loading:true, 
    }
    componentWillMount() {
      const { getProducts, } = this.props;
      getProducts();
    }

    handleLogout = (event) => {
      event.preventDefault();
      const token = localStorage.getItem("token");
      if(token){
       if(confirm("Are you sure you want to logout?")){
        localStorage.clear()
        this.setState({ redirect: true, })
       }
      }
    }
    onCloseModal = () => {
      this.setState({ open: false})
    }

    showRegisterForm = (event) => {
      event.preventDefault();
      this.setState({ open: true });
    }

    render() {
        const token = localStorage.getItem("token");
        const role = localStorage.getItem("userRole");
        const { products, } = this.props;
        const {redirect, open, loading, } = this.state;

        if (!token) {
            return (
              <Redirect to="/" />
            );
        }
        if(redirect){
          return (
            <Redirect to="/" />
          );
        }

        return (
          <div>
            <Navigation onLogout={this.handleLogout}/>
            <div className="row">
              <Dashboard products={products} />
            </div>
            {
              role === 'admin' ? 
              <CornerMenu onRegister={this.showRegisterForm} />:''
            }
            <UserAddView open={open} onClose={this.onCloseModal} />
          </div>

        );
    }
}

const mapStateToProps = state => ({
    products: state.productListReducer.products,

});


export default connect(mapStateToProps, { getProducts, })(DashboardView);
