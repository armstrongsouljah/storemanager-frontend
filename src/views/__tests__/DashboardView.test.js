import React from "react";
import { shallow, } from "enzyme";
import DashboardView, {DashboardView as A2} from '../DashboardView';
const props = {
    products: {
        length: 2,
    },
    getProducts: jest.fn(),
}

describe("<DashboardView/>", () => {
    it("should render without crashing", () => {
        const wrapper = new DashboardView(props);
        expect(wrapper).toMatchSnapshot();
    });
    it("should populate products on load", () => {
       const wrapper = shallow(<A2 {...props}/>);
       wrapper.instance().getProducts = jest.fn();
       wrapper.instance().getProducts()
       expect(wrapper.instance().props.products.length).toBeGreaterThan(0)
    });
    it("should should call setState on handleLogout", () => {
        const wrapper = shallow(<A2 {...props} />)
        wrapper.instance().handleLogout = jest.fn()
        wrapper.instance().handleLogout({
            preventDefault(){}
        })
        wrapper.instance().setState = jest.fn();
        global.confirm = jest.fn()
        global.confirm("are you sure")
        wrapper.instance().handleLogout({ preventDefault(){}})
        wrapper.instance().setState({redirect: true})

    })
});
