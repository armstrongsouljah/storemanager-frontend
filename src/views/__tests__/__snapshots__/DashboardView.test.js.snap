// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardView/> should render without crashing 1`] = `
Connect {
  "context": undefined,
  "indirectRenderWrappedComponent": [Function],
  "props": Object {
    "getProducts": [MockFunction],
    "products": Object {
      "length": 2,
    },
  },
  "refs": Object {},
  "selectChildElement": [Function],
  "selectDerivedProps": [Function],
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
